articles/regularization/index.md

```diff
@@ -0,0 +1,84 @@
+Let’s use a linear regression equation to explain regularization further.
+
+$$ Y = \beta_0 + \beta_1 X_1 + \beta_2 X_2 + … + \beta_p X_p $$
+
+$Y$ represents the value that is to be predicted. $\beta$ stands for the coefficient estimates for various predictors $(X)$. It represents the weights or magnitudes assigned to various predictors.
+A loss function is involved in the fitting process. This loss function is referred to as the residual sum of squares (RSS). Linear models attempt to optimize the coefficients to minimize the loss function. This is shown by the equation for the loss function below.
+
+$$ RSS = \sum\limits_{i=1}^n( y_i – \beta_0 - \sum\limits_{j=1}^p {\beta_j x_ij )^2} $$
+
+Based on the training data, the coefficients shall be adjusted. If there is the existence of noise in the training data, the approximated coefficients shall not generalize well to unseen data. Regularization comes into play and shrinks the learned estimates towards zero.
+
+### Lasso Regression
+
+Lasso regression is a regularization technique to reduce model complexity. It is also known as L1 regularization. Lasso stands for Least Absolute Shrinkage and Selector Operator. Let’s look at the equation below.
+
+$$ \sum\limits_{i=1}^n( y_i – \beta_0 - \sum\limits_{j=1}^p {\beta_j x_ij )^2} + \lambda \sum\limits_{j=1}^p \mid \beta_j \mid = RSS + \lambda \sum\limits_{j=1}^p \mid \beta_j \mid $$
+
+We note that it is a variation of the loss function equation, with the introduction of a penalty term. To penalize high coefficients, lasso uses absolute values of beta $ (\mid \beta\mid) $. Whereas, as we shall learn, ridge regression uses the squares of beta $(\beta^2)$ for a similar purpose.
+Lasso shrinks the regression coefficients to regularize the model parameters. Unlike ridge regression, as we shall cover later, lasso can reduce some regression coefficients to zero. This is particularly important when it comes to feature selection.
+The feature selection process is attributed to the ability of lasso to reduce some regression coefficients to zero. It comes after the regression coefficients are shrunk. Every non-zero regression coefficient is selected for use in the model. Since some features are eliminated, less important features will not feature in the final model. This greatly assists in minimizing prediction errors.
+
+Lasso also improves the prediction accuracy of models. The shrinking of coefficients minimizes the bias and reduces the variance of models. A context that favors the use of lasso is when we have a high number of features and a small amount of data. The tuning parameter $\lambda$ controls the shrinkage. From the equation, when $\lambda$ is zero, the equation is reduced to the linear regression loss function equation. The greater the value of $\lambda$, the greater the reduction of the coefficients towards zero.
```

Comment:
```suggestion
Lasso also helps improving the prediction accuracy of the models. The shrinking of coefficients minimizes the bias and reduces the variance of models. A context that favors the use of lasso is, when we have a small dataset with more number of features. The tuning parameter $\lambda$ controls the shrinkage. From the equation, when $\lambda$ is zero, the equation is reduced to the linear regression loss function equation. The greater the value of $\lambda$, the greater the reduction of the coefficients towards zero.
```

`The shrinking of coefficients minimizes the bias and reduces the variance of models.`

Fyi, Bias and Variance are inversely proportional.